src/inmanta/data/__init__.py

```diff
@@ -4863,7 +4863,7 @@ async def copy_resources_from_unchanged_resource_set(
                 destination_version,
                 updated_resource_sets | deleted_resource_sets,
             )
-            return {record["resource_id"] for record in result}
+            return {record["resource_id"]: record["resource_set"] for record in result}
```

Comment:
I noticed mypy complains about several return values e.g. `src/inmanta/data/__init__.py:5286: error: Incompatible return value type (got "List[object]", expected "List[str]")  [return-value]` [here](https://github.com/inmanta/inmanta-core/blob/b1cd58a6351a467b6c0c69ef560d1d1f15172d57/src/inmanta/data/__init__.py#LL5286C22-L5286C22) 
Should we cast before returning ?
```suggestion
            return {str(record["resource_id"]): str(record["resource_set"]) for record in result}

```

Reply:
That's a common issue. Data that comes from the database always has the type object, so you need to cast it yourself.